department-of-veterans-affairs/vets-website
# WIP: Terms of Use | API interaction

## Summary
This PR adds the API integration for users to integrate with the Terms of Use actions of Accepting or Declining.
*Note: this will not include the auto-signout after decline button interaction nor redirecting to the `/decline` page

## Related issue(s)


## Testing done
- Unit testing + manual testing

## Screenshots


## What areas of the site does it impact?
This will affect just the Terms of Use page `/terms-of-use` in staging & below

## Acceptance criteria

### Quality Assurance & Testing


### Error Handling


### Authentication


### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:


## Requested Feedback
1. Pull down latest in `vets-api` and start the server
2. Pull down this branch `tou/flipper`
3. Run `yarn` then `yarn watch` 
4. Navigate to `localhost:3001/terms-of-use`
    1. Confirm there are no buttons (Accept or Decline) to click because you are not signed in
5. Sign into VA.gov using any available CSP
6. After authenticating navigate to `localhost:3001/terms-of-use`
7. Open up your DevTools on the Network tab
    1.  Click Accept button confirm the API URL includes the `terms_of_use_agreements/v1/accept`
    2. Click Decline button confirm the API URL includes the `terms_of_use_agreeemnts/v1/decline`
8. Stop the `vets-api` server by running Ctrl + C
9. Try to click the Accept/Decline button
10. Confirm there is an error message asking for them to try again.
